pkg/apis/controlplane/v1beta1/types.go

```diff
@@ -298,3 +298,18 @@ type NetworkPolicyStats struct {
 	// The stats of the NetworkPolicy.
 	TrafficStats statsv1alpha1.TrafficStats `json:"trafficStats,omitempty" protobuf:"bytes,2,opt,name=trafficStats"`
 }
+
+// NetworkPolicyQueryFilter is used to filter the result while retrieve network policy
+// Not settled attributes means match all.
+// e.g SourceType = "" means all type network policy will be retrieved
+// Can have more attributes in future if more args are required
+type NetworkPolicyQueryFilter struct {
```

Comment:
I haven't reviewed the whole PR yet, but I'm wondering why `NetworkPolicyQueryFilter` is included in the `controlplane` API given that it seems to be used exclusively on the server side in the agent. `NewFilterFromURLQuery` builds an instance of this struct based on the URL query, we do not get a serialized json `NetworkPolicyQueryFilter` object from the client (antctl). So unless you have future plans for this struct, it seems weird to include it as part of this versioned API.

Reply:
Thanks for your advice.
I was thinking that since name+namespace will no longer have the ability to specify a network policy. We might need this `NetworkPolicyQueryFilter` on the server-side or any places that need to retrieve or locate a network policy in the future. 
But yeah now it's totally fine and making more sense to move this class into `pkg/agent/controller/networkpolicy/cache.go` or other places that you suggested.

Reply:
Until we actually need it in the API, yes I recommend not including it in the API

Reply:
Got it